### PR TITLE
Test multiple collaborators on share on a single share

### LINKED ISSUE #2001: acceptance tests for adding multiple collaborators in one step
its possible to add multiple collaborators at once, we need tests for that

### PR DESCRIPTION
**Pull request by skshetry**
## Description

This PR adds acceptance tests for adding multiple collaborators (users and groups) on sharing a file/folders.

## Related Issue

- Fixes #2001 

## How Has This Been Tested?

- locally
- :robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5236/

<p>





</p></details>

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 1]
```diff
@@ -179,3 +179,110 @@ Feature: Sharing files and folders with internal groups
...
+    And user "user2" has been added to group "grp2"
+    And user "user1" has logged in using the webUI
+    When the user opens the share dialog for file "lorem.txt" using the webUI
+    And the user selects the following collaborators for the share as "<role>" with "<extra-permissions>" permissions:
+      | collaborator | type  |
+      | grp1         | group |
+      | User Three   | user  |
+      | grp2         | group |
+    And the user removes "grp1" as a collaborator from the share
+    And the user removes "User One" as a collaborator from the share
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: "User One" was not collaborator, so can we remove that user?

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 2]
```diff
@@ -192,6 +192,25 @@ When('the user shares file/folder {string} with user {string} as {string} with p
...
+  client.page.FilesPageElement.sharingDialog().selectRoleForNewCollaborator(role)
+  await client.page.FilesPageElement.sharingDialog().selectPermissionsOnPendingShare(permissions)
+})
+
+When('the user removes {string} as a collaborator from the share', function (user) {
+  return client.page.FilesPageElement.sharingDialog().removePendingCollaboratorForShare(user)
+})
+
+When('the user shares with the selected collaborators', function () {
+  return client.page.FilesPageElement.sharingDialog().confirmShare().waitForOutstandingAjaxCalls()
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/sharingContext.js]:
```suggestion
  return client.page.FilesPageElement.sharingDialog()
  .confirmShare()
  .waitForOutstandingAjaxCalls()
```

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 3]
```diff
@@ -20,56 +20,84 @@ module.exports = {
...
+     */
+    selectCollaboratorForShare: async function (sharee, shareWithGroup = false) {
+      await this.enterAutoComplete(sharee)
         .waitForElementPresent({
           selector: '@sharingAutoCompleteDropDownElements',
           abortOnFailure: false
         }, (result) => {
           if (result.value === false) {
-            // sharing dropdown was not shown
-            console.log('WARNING: no sharing autocomple dropdown found, retry typing')
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: why deleting this line?

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 4]
```diff
@@ -20,56 +22,88 @@ module.exports = {
...
-     * @param {string} role
-     * @param {string} permissions
      */
-    shareWithUserOrGroup: async function (sharee, shareWithGroup = false, role, permissions) {
-      this.enterAutoComplete(sharee)
+    removePendingCollaboratorForShare: function (sharee) {
+      const newCollaboratorXpath = util.format(this.elements.newCollaboratorItems.selector, sharee)
+      const removeCollaboratorBtnXpath = newCollaboratorXpath + this.elements.newCollaboratorRemoveButton.selector
+
+      return this.useXpath().click(removeCollaboratorBtnXpath)
```
Comment (HariBhandari07) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: ```return this.useXpath().click(removeCollaboratorBtnXpath).useCss()```

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js (hunk 3 above)
Comment (skshetry) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: Oops, fixed it

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5250/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5250/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5250/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5267/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5268/

<p>














</p></details>

**Comment by skshetry** [comment]
Too many random failures in phoenix. :( Restarted

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5270/

<p>





</p></details>

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 5]
```diff
@@ -78,7 +112,7 @@ module.exports = {
      * @param {String} role
      */
     selectRoleForNewCollaborator: function (role) {
-      role = role.replace(' ', '')
+      role = _(role).chain().toLower().startCase().replace(/\s/g, '').value()
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: can you explain what happens here? maybe add a comment

Comment (skshetry) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: I thought it was pretty readable. Any suggestions to make it more readable?

Comment (skshetry) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: I'd hate comment here. Would this work?
```suggestion
      const titleCase = (string) => _(role).chain().toLower().startCase().value()
      role = _.replace(titleCase(role), /\s/g, '')
```

**Pull request merged**